### ISSUE TITLE
New features - offering to share

### ISSUE THREAD

**Issue by ghaecker** [StephanWagner/jBox#106]
Thank you for your excellent work on this plugin.  I'm using it in one of my projects.

Rather than bug you with feature requests, I forked your project to alter some behaviors in jBox.  My implementation doesn't change any default jBox behavior.  They are optional settings, and they're fairly lightweight.

**jBox.useTrueModal** &mdash; a new Boolean property.  It applies to jBoxes with overlay:true.  When set to true, ideally in document.ready() before instantiating any jBoxes, it stacks zIndex for overlays, instead of having them all assigned 9999.  This property makes jBoxes with overlays behave as true modal windows by covering other jBoxes.  They're excluded from dragOver behavior, but have a similar behavior on open, forcing them (and their overlay) to the top of the zIndex.  I know zIndex could be managed manually in a project, but it can get messy quickly when there are a log of modals.

**options.holdPosition** &mdash; a new Boolean option for draggable jBoxes.  When true, it works similar to setting the 2nd arg of setContent() to true for dimension changes.  It differs, though, in that it also keeps the draggable jBox (if it's been moved from its original position) from reverting back to its original position on window resize and it constrains the jBox to the viewport, as available space allows.

Anyway, if you have any interest in these features I'll be happy to open a PR.

**Repository:** https://github.com/ghaecker/jBox.git

**Branches:**  trueModal and holdPosition

**Comment by StephanWagner** [comment]
Hi Glenn,

holdPosition looks very nice. That definitely is a nice addition. I'll have a more detailed look into useTrueModal too. As you mentioned, you could just achieve this by using different zIndex. But maybe some automatism would be nice too. Feel free to create pull requests for those.

Thanks!
Stephan

--
Stephan Wagner
Am Gojenboom 23A
22111 Hamburg

+49 176 45783473
https://stephanwagner.me

**Comment by ghaecker** [comment]
Thanks for taking a look.  I'll create the pull requests.

-- Glenn

**Comment by ghaecker** [comment]
Hi Stephan,

Based on your recent zIndex: 'auto' work, I've switched true modal handling in my fork from my original implementation to use yours.  There are, however, a few issues I addressed.  Thank you for the time you spent on this feature.

* Line 262: keeping original default of 10000 works fine.
* Line 265: += 2 is more moderate than += 10 and works fine.
* Line 355: added condition to exclude from dragOver if jBox is set to adjustZIndexOnOpen and has an overlay.  Such jBoxes are already at the top of the zIndex - no need to bump them higher.
* Line 356: the original += 1 increment still works fine for dragOver.
* Lines 355-359, 388: eliminated separate jBox.zIndexMaxDragover property; changed it back to jBox.zIndexMax, so both incrementing operations are aware of the same max value.  As it was, jBox modal with overlay, could still open under other jBoxes.  The two properties were not adequately coordinated.

I've tested this pretty thoroughly and think it's behaving as you intended, as well as in a manner that meets my objective.

My updates to the zIndex implementation are in a separate branch, zIndexAuto, in my fork, in case you'd like me to create a PR.

https://github.com/ghaecker/jBox.git
